### DISCUSSION TITLE
Current status of Tomcat and Jetty adapters

### DISCUSSION THREAD

**Discussion by mabartos** [keycloak/keycloak#8932] [Archive]
Hello!

I'd like to discuss here some aspects related to _Keycloak Java_ adapters, because I think, it'd be great to improve them. At this moment, in our codebase, we support particular Java adapters for _Tomcat_, _Jetty_ and so on. However, the supported versions are quite old and it'd be good to upgrade them and support more modern application servers.

For instance, we provide adapters for _Jetty 9.2_, _9.3_ and _9.4_, but _Jetty 10.x_ and even _Jetty 11.x_ is released at this moment. IMO, it'd worth it to get rid of the _9.2_ and _9.3_ (deprecated) Jetty adapters and rather support more recent versions. Of course, it's necessary to find out, if it's really worth the effort and if the community would be glad to have it. At least, from QE perspective, it'd be great to test the newest application servers, because we don't know if our current adapters really works with them (assume not). 

The same issue is similar for the _Apache Tomcat_, where we provide _7_, _8_ and _9_ version. However, _Tomcat 7_ is deprecated and it'd be good to consider if we have some needs to support it. There's also a new _Tomcat 10_ release and we don't know if our current _Tomcat_ adapter works with it. There's also extended version of the _Apache Tomcat_ application server with name _TomEE_. However, everything depends largely on customer requests and needs, right?

I'd like to find out the current status of _Keycloak_ adapters and their planned future. IMO, it'd be great to keep them alive and bring them up to date with the times.

WDYT? cc: @stianst, @hmlnarik, @mhajas, @pdrozd, @miquelsi

**Comment by stianst** [comment]
WildFly has recently introduced native support with OIDC, Spring Security has done the same. That effectively means it makes no sense for us to continue the effort with WildFly and Spring adapters. Tomcat and Jetty adapters do have some use, but it is quite limited, and we can't really justify the effort required to keep them alive for a very small user base.

In terms of modernising the adapters, there is also a fairly big gap between what we support today on the adaptor side and what we support on the server side around OIDC and OAuth.

**Comment by optyfr** [comment]
any chance to have a keycloak adapter for jetty 10 ?

**Reply by stianst** [reply]
Afraid not, we're phasing out all adapters and won't update them to support new versions.

**Comment by stianst** [comment]
Rejecting this as we are phasing out all adapters, see https://www.keycloak.org/2022/02/adapter-deprecation.html for more details.
